### PR TITLE
Better two-dimensional turbulence Docs example

### PR DESCRIPTION
**Pull request by navidcy**
This PR uses the latest of the time-stepping and advection schemes to produces a two-dimensional turbulence example that resembles two-dimensional turbulence :)

2d_turbulence_vorticity

### CODE REVIEW

**Comment by navidcy** [comment]
@ali-ramadhan, @glwagner, @oceananigans-consortium: Do I win anything for 1000 issue/PR? :)

**Comment by glwagner** [comment]
> Do I win anything for 1000 issue/PR? :)

OMG!

How about 1000 hugs?

You should probably frame this and put it on your wall too.

File: examples/two_dimensional_turbulence.jl
Comment (glwagner) [inline examples/two_dimensional_turbulence.jl:91]: @navidcy do you mind if I convert this example to run the simulation and post-process _afterwards_ (like in `langmuir_turbulence.jl`)? This is an outstanding issue #676

**Comment by glwagner** [comment]
Looks like this example ran afoul of #941

File: examples/two_dimensional_turbulence.jl
Comment (navidcy) [reply examples/two_dimensional_turbulence.jl:91]: I’ll do it as an exercise:)

Comment (navidcy) [reply examples/two_dimensional_turbulence.jl:91]: I’ll do it as an exercise:)

**Comment by navidcy** [comment]
What do you mean “ran afoul”?

**Comment by glwagner** [comment]
> What do you mean “ran afoul”?

https://www.dictionary.com/browse/run--afoul--of#:~:text=Also%2C%20run%20foul%20of.,transferred%20to%20non%2Dnautical%20usage.

**Comment by navidcy** [comment]
> > Do I win anything for 1000 issue/PR? :)
> 
> OMG!
> 
> How about 1000 hugs?
> 
> You should probably frame this and put it on your wall too.

1000 hugs are very welcome!
Also an Oceananigans.jl coffee mug would be cool! ;)

**Comment by ali-ramadhan** [comment]
Oooh it's so pretty now!

> Also an Oceananigans.jl coffee mug would be cool! ;)

Haha now we need a logo.

**Comment by ali-ramadhan** [comment]
If only this PR came in a week earlier haha: https://www.youtube.com/watch?v=GCj4xHUGZ0g&feature=youtu.be&t=2366

**Comment by navidcy** [comment]
I don't understand why tests fail; any insight?

File: examples/two_dimensional_turbulence.jl
Comment (ali-ramadhan) [inline examples/two_dimensional_turbulence.jl:19]:
```suggestion
        grid = RegularCartesianGrid(size=(128, 128, 1), halo=(2, 2, 2), extent=(2π, 2π, eps(Float64))),
```

Ah `CenteredFourthOrder` needs halos of size 2. This should fix the failing test.

This should be made more automatic and user-friendly. See #941.

Comment (ali-ramadhan) [reply examples/two_dimensional_turbulence.jl:19]: Hmmm, I wonder if it's a good use to use `eps(Float64)` in a beginner example. Maybe we should wait until #1003 is merged to get rid of the `eps(Float64)` and make the example more user-friendly?

Comment (navidcy) [reply examples/two_dimensional_turbulence.jl]: oh that was a mistake! I didn't mean to push `eps(Float64)`!!

**Comment by glwagner** [comment]
> If only this PR came in a week earlier haha: https://www.youtube.com/watch?v=GCj4xHUGZ0g&feature=youtu.be&t=2366

This is an important example!

**Review by glwagner — APPROVED** [review]

**Pull request merged**
